### DISCUSSION TITLE
Pagination by the left side only when using join with contains_eager in query

### DISCUSSION THREAD

**Discussion by alexted** [sqlalchemy/sqlalchemy#8228] [Usage Questions] [answered]
Hello! My problem is somewhat similar to the one discussed in https://github.com/sqlalchemy/sqlalchemy/discussions/7951 but I am interested in the following:
I'm using **join** with **contains_eager** and also **limit** and **offset** in my query, and the pagination result was unexpected for me. Previously (when i use sub-queries) the limit was counted only by parent (left side) objects, but now I get a parent+child assembly (i saw it in the raw result), and the **limit** gives me less parent objects than I need/await! 
Tell me pls what to do? Do pagination in code after getting orm-objects?

**Comment by CaselIT** [comment, marked as answer]
Hi

you could probably paginate the parent, make a subquery and join that with the child. Something like the folowwing
```sql
select *
from (select * from parent limit 10) as p
left join child on ...
```

**Reply by alexted** [reply]
Could you tell me how to do this with an ORM? 
Here is my current query:
```python
select(Browser)
.outerjoin(Browser.versions)
.where(or_(Browser.name.ilike(f"{name}%"), BrowserVersion.name.ilike(f"{version_name}%")))
.limit(filters.limit).offset(filters.offset).order_by(asc(Browser.id))
.options(contains_eager(Browser.versions))
.execution_options(populate_existing=True)
```

**Reply by CaselIT** [reply]
it's pretty similar to sql, but you need to use aliased on Browser for the subquery
```py
br_limit = aliased(
  Browser, 
  select(Browser)
  .where(or_(Browser.name.ilike(f"{name}%"), BrowserVersion.name.ilike(f"{version_name}%")))
  .limit(filters.limit).offset(filters.offset).order_by(asc(Browser.id))
  .subquery()
)
q = select(br_limit).outerjoin(br_limit.versions).options(
  contains_eager(br_limit.versions)).execution_options(populate_existing=True)
values = session.scalars(q).unique().all()
```

**Reply by alexted** [reply]
Thank you so much for your help!
